File: front/minhas-despesas/src/models/despesa-pagamento-dto.ts
import { DespesaDto } from "./despesa-dto";

export interface DespesaPagamentoDto{
    despesaPagamentoId: string;
    despesaId: string;
    despesa: DespesaDto;
    dataPagamento: Date;
    valor: number;
    observacao?: string;
}

File: back/MinhasDespesas/MinhasDespesas.Models/Dtos/DespesaPagamentoDto.cs
namespace MinhasDespesas.Models.Dtos;

public class DespesaPagamentoDto
{
    public Guid DespesaPagamentoId { get; set; }
    public Guid DespesaId { get; set; }
    public DespesaDto Despesa { get; set; }
    public DateTime DataPagamento { get; set; }
    public Double Valor { get; set; }
    public string? Observacao { get; set; }
}